### ISSUE TITLE
(🐞) `TypeGuard`s incorrectly become the guard type when passed to a `TypeVar`

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#17114]
https://github.com/python/mypy/labels/topic-typeguard https://github.com/python/mypy/labels/topic-type-variables
```py
from typing import TypeGuard, TypeVar, Callable

T = TypeVar("T")

def guard(x: object) -> TypeGuard[str]: 
    return True

def f(fn: Callable[[object], T]) -> T: ...
a = f(guard)  # error: Argument 1 to "f" has incompatible type "Callable[[object], TypeGuard[str]]"; expected "Callable[[object], str]"  [arg-type]
reveal_type(a) # str
```

**Comment by JelleZijlstra** [comment]
This looks to be fixed on master, probably by #17071. It now outputs:

```
<string>:11: error: Missing return statement  [empty-body]
<string>:13: note: Revealed type is "Any"
Found 1 error in 1 file (checked 1 source file)
```

**Comment by KotlinIsland** [comment]
@JelleZijlstra That's great that it doesn't have the false error any more, but this is certainly not fixed, it's `Any` now, not `bool`. I guess I'll raise a new ticket.
